external/fv3fit/fv3fit/sklearn/_random_forest.py

```diff
@@ -274,10 +278,12 @@ def _predict_on_stacked_data(self, stacked_data):
         )
 
     def predict(self, data):
-        stacked_data = stack_non_vertical(
-            safe.get_variables(data, self.input_variables)
+        input_data = safe.get_variables(data, self.input_variables)
+        stacked_data = (
+            stack(input_data)
+            if self.predict_columns is False
+            else stack_non_vertical(input_data)
```

Comment:
I think you could just delete the `stack_non_vertical` function and just call `stack` in either case with different unstacked_dims args. `stack_non_vertical` is pretty old and basically only exists to do some checks in the case that `pfull` is the vertical dimension which was only the case in the very early days of our ML workflow.

Reply:
Sure. `stack_non_vertical` is called in the novelty detectors etc. but switching it to `stack` isn't hard.